sdks/java/io/iceberg/src/test/java/org/apache/beam/sdk/io/iceberg/IcebergIOIT.java

```diff
@@ -307,4 +314,38 @@ public void testWritePartitionedData() {
     assertThat(
         returnedRecords, containsInAnyOrder(INPUT_ROWS.stream().map(RECORD_FUNC::apply).toArray()));
   }
+
+  @Test
+  public void testStreamingWrite() {
```

Comment:
I suggest a couple more tests where the user has set up their PCollection differently, like if it started out with accumulating mode, or if they set a weird trigger in the middle of their pipeline.

Reply:
Added a test with fixed windows and accumulating mode. Let me know if there's anything particular we should test for